appsec/events/block.go

```diff
@@ -0,0 +1,22 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+// Package events provides the types and interfaces for the appsec event system.
+// User-facing events can be returned by the appsec package to signal that a request was blocked.
+// Handling these events differently than other errors is crucial to not leak information to an attacker.
+package events
+
+var _ error = (*SecurityBlockingEvent)(nil)
+
+// SecurityBlockingEvent is an event that signals that a request was blocked by the WAF.
+// It should be handled differently than other errors to avoid leaking information to an attacker.
+// If this error was returned by native types wrapped by dd-trace-go, it means that a 403 response will be written
+// by appsec middleware (or any other status code defined in DataDog's UI). Therefore, the user should not write a
+// response in the handler.
```

Comment:
```suggestion
// BlockingSecurityEvent is the error type returned by function calls blocked by appsec.
// Even though appsec takes care of responding automatically to the blocked requests, it
// is your duty to abort the request handlers that are calling functions blocked by appsec.
// For instance, if a gRPC handler performs a SQL query blocked by appsec, the SQL query
// function call gets blocked and aborted by returning an error of type SecurityBlockingEvent.
// This allows you to safely abort your request handlers, and to be able to leverage errors.As if
// necessary in your Go error management logic to be able to tell if the error is a blocking security
// event or not (eg. to avoid retrying an HTTP client request).
```

+ this complex comment is going to be clearer with a great and clear example. Can you add two examples please:
1. A simple one showing the fact it is not required to use errors.As in the simple and commong case of "if err != nil { return }
2. A more advanced one where there is some logic with the error of a HTTP client request (http.Post(something)) that gets retried if this is not an appsec event, or otherwise returns immediately.